src/handlers/tsonAsyncIterable.ts

```diff
@@ -23,26 +23,24 @@ export const tsonAsyncIterator: TsonAsyncType<
 	async: true,
 	deserialize: (opts) => {
 		return (async function* generator() {
-			try {
-				for await (const value of opts.stream) {
-					switch (value[0]) {
-						case ITERATOR_DONE: {
-							return;
-						}
+			let next: ReadableStreamReadResult<SerializedIteratorResult>;
+			while (((next = await opts.reader.read()), !next.done)) {
```

Comment:
This could prob be prettier :D

Reply:
TBH, `while(true)` and `if(next.done) return` would have worked fine, but then eslint doesn't like `while(true)` so I went with this. We can `eslint-disable-next-line` instead if you prefer.